src/Styleguide/Elements/Button.tsx

```diff
@@ -1,40 +1,126 @@
-import React, { SFC } from "react"
+import React, { Component, ReactNode } from "react"
 import styled from "styled-components"
-import { Sans } from "@artsy/palette"
+import { Sans, TypographyProps } from "@artsy/palette"
 import { pick } from "lodash/fp"
 
 import {
   border,
+  borderColor,
+  BorderColorProps,
+  BorderProps,
   borderRadius,
+  BorderRadiusProps,
   color,
+  ColorProps,
   maxWidth,
   space,
+  SpaceProps,
   textAlign,
+  TextAlignProps,
   width,
+  WidthProps,
 } from "styled-system"
 
-export const Button: SFC = ({ children, ...rest }) => {
-  const textProps = pick(["color", "size", "weight"], rest)
+enum ButtonVariant {
+  primary = "primary",
+  secondary = "secondary",
+  outline = "outline",
+}
+
+enum ButtonSize {
+  large = "large",
+  default = "default",
+  small = "small",
+  xsmall = "xsmall",
+}
+
+export interface ButtonProps extends ButtonBaseProps {
+  children: ReactNode
+  size?: any // FIXME, ButtonSize
+  variant?: any // FIXME, ButtonVariant?
```

Comment:
@alloy - I'm having trouble assigning a possible enum to a prop. When I set 
```
variant?: ButtonVariant`
```
and then try to use the button like 
```
<Button variant='outline' />
```
it throws this error: 

<img width="442" alt="screen shot 2018-06-08 at 12 50 01 am" src="https://user-images.githubusercontent.com/236943/41145772-fe3bb3d0-6ab5-11e8-8297-4a28d28f47b1.png">

Not quite sure I understand how enums / options work.

Reply:
The error may be that you are passing a string value where an enum key is expected. You should probably just use a string union instead of an enum. Is there any value in defining the enum as is done anyways?

Reply:
My thinking around enums is that you can also use them as values, so in situations like this where there will eventually be a large number of variants it keeps the drift level to a minimum. I remember luc doing something similar recently and it felt clean.

This totally answers the question tho, I was thinking it was looking for the value when really it is looking for the key.